core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotService.scala

```diff
@@ -152,28 +165,37 @@ class SnapshotService(protected val ctx: RawlsRequestContext,
         throw new RawlsExceptionWithErrorReport(ErrorReport(other))
     }
 
+  def enumerateSnapshotsByWorkspaceName(workspaceName: WorkspaceName,
+                                        offset: Int,
+                                        limit: Int,
+                                        referencedSnapshotId: Option[UUID] = None
+  ): Future[SnapshotListResponse] =
+    getV2WorkspaceContextAndPermissions(workspaceName,
+                                        SamWorkspaceActions.read,
+                                        Some(WorkspaceAttributeSpecs(all = false))
+    ).flatMap(workspaceContext =>
+      enumerateSnapshots(workspaceContext.workspaceIdAsUUID, offset, limit, referencedSnapshotId)
+    )
+
   /**
     * return a given page of snapshot references from Workspace Manager, optionally returning only those
     * snapshot references that refer to a supplied TDR snapshotId.
     *
-    * @param workspaceName the workspace owning the snapshot references
+    * @param workspaceId the id of the workspace owning the snapshot references
     * @param offset pagination offset for the list
     * @param limit pagination limit for the list
     * @param referencedSnapshotId the TDR snapshotId for which to return matching references
     * @return the list of snapshot references
     */
-  def enumerateSnapshots(workspaceName: WorkspaceName,
+  def enumerateSnapshots(workspaceId: UUID,
                          offset: Int,
                          limit: Int,
                          referencedSnapshotId: Option[UUID] = None
   ): Future[SnapshotListResponse] =
-    getV2WorkspaceContextAndPermissions(workspaceName,
-                                        SamWorkspaceActions.read,
-                                        Some(WorkspaceAttributeSpecs(all = false))
-    ).map { workspaceContext =>
+    getV2WorkspaceContextAndPermissionsById(workspaceId.toString, SamWorkspaceActions.read).map { workspaceContext =>
```

Comment:
this can/should also specify `attributeSpecs = Some(WorkspaceAttributeSpecs(all = false)`